modules/renter.go

```diff
@@ -50,12 +55,154 @@ type Allowance struct {
 // DownloadInfo provides information about a file that has been requested for
 // download.
 type DownloadInfo struct {
-	SiaPath     string    `json:"siapath"`
-	Destination string    `json:"destination"`
-	Filesize    uint64    `json:"filesize"`
-	Received    uint64    `json:"received"`
-	StartTime   time.Time `json:"starttime"`
-	Error       string    `json:"error"`
+	SiaPath     string         `json:"siapath"`
+	Destination DownloadWriter `json:"destination"`
+	Filesize    uint64         `json:"filesize"`
+	Received    uint64         `json:"received"`
+	StartTime   time.Time      `json:"starttime"`
+	Error       string         `json:"error"`
+}
+
+// DownloadWriter provides an interface which all output writers have to implement.
+type DownloadWriter interface {
+	WriteAt(b []byte, off int64) (int, error)
+	Location() string
+}
+
+// DownloadBufferWriter is a buffer-backed implementation of DownloadWriter.
+type DownloadBufferWriter struct {
```

Comment:
I'd go ahead and move all of these implementations directly into the renter. I don't see any reason for them to be defined at the modules layer, especially b/c I imagine that there will be a lot of implementation specific context that we add to them later.